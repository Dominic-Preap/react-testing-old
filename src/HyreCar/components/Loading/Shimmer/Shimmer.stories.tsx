import * as React from 'react';

import { boolean, number } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';

import { Shimmer } from './Shimmer';

storiesOf('HyreCar|Components/Loading', module).add('Shimmer', () => (
  <Shimmer width={number('Width', 100)} height={number('Height', 20)} round={boolean('Round', false)} />
));
