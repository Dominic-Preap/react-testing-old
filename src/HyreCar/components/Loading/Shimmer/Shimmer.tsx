import styled, { keyframes } from 'styled-components';

const placeholder = keyframes`
  0% {
    background-position: -468px 0;
  }
  100% {
    background-position: 468px 0;
  }
`;

interface Props {
  /**
   * Shimmer height
   *
   * @default 50
   */
  height?: string | number;
  /**
   * Shimmer width
   *
   * @default 50
   */
  width?: string | number;
  /**
   * if `true`, set border-radius to 100%
   *
   * @default false
   */
  round?: boolean;
  /**
   * Overrides the existing style
   */
  style?: React.CSSProperties;

  /**
   * Overrides the existing style with className
   */
  className?: string;
}

export const Shimmer: React.FC<Props> = styled.div<Props>`
  /* ! WARNING: Do not re-arrange the css or the animation won't work */
  background: #f6f7f8;
  background-image: linear-gradient(to right, #f6f7f8 0%, #edeef1 20%, #f6f7f8 40%, #f6f7f8 100%);
  background-repeat: no-repeat;
  background-size: 800px 300px;
  display: inline-block;
  position: relative;
  animation: ${placeholder} 1s linear infinite forwards;

  /* Props define here */
  border-radius: ${props => (props.round ? '100%' : '0')};
  width: ${props => (typeof props.width === 'number' ? (props.width || 50) + 'px' : props.width)};
  height: ${props => (typeof props.height === 'number' ? (props.height || 50) + 'px' : props.height)};
`;
