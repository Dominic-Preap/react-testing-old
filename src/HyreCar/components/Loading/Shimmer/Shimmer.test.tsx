import * as React from 'react';
import { create } from 'react-test-renderer';

import { Shimmer } from './Shimmer';

describe('Shimmer component', () => {
  it('should render properly', () => {
    expect(create(<Shimmer />).toJSON()).toMatchSnapshot();
  });
});
