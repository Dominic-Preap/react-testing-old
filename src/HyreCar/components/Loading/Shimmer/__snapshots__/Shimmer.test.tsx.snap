// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shimmer component should render properly 1`] = `
.c0 {
  background: #f6f7f8;
  background-image: linear-gradient(to right,#f6f7f8 0%,#edeef1 20%,#f6f7f8 40%,#f6f7f8 100%);
  background-repeat: no-repeat;
  background-size: 800px 300px;
  display: inline-block;
  position: relative;
  -webkit-animation: iiNPHT 1s linear infinite forwards;
  animation: iiNPHT 1s linear infinite forwards;
  border-radius: 0;
}

<div
  className="c0"
/>
`;
